hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/AvroConvertor.java

```diff
@@ -180,7 +180,7 @@ public GenericRecord withKafkaFieldsAppended(ConsumerRecord consumerRecord) {
     recordBuilder.set(KAFKA_SOURCE_OFFSET_COLUMN, consumerRecord.offset());
     recordBuilder.set(KAFKA_SOURCE_PARTITION_COLUMN, consumerRecord.partition());
     recordBuilder.set(KAFKA_SOURCE_TIMESTAMP_COLUMN, consumerRecord.timestamp());
-    recordBuilder.set(KAFKA_SOURCE_KEY_COLUMN, consumerRecord.key().toString());
+    recordBuilder.set(KAFKA_SOURCE_KEY_COLUMN, String.valueOf(consumerRecord.key()));
```

Comment:
So the null key is represented as a `null` constant string.

Reply:
yes correct !

Reply:
Hi Danny, have refactored a bit to handle null kafka key, please re-review